[2025-04-18 04:08:29,992][__main__][INFO] - Instantiating trainer: src.trainers.sae_trainer.SAETrainer
[2025-04-18 04:08:31,893][src.trainers.base_trainer][INFO] - Using device: cuda
[2025-04-18 04:08:31,893][__main__][INFO] - Trainer instantiated successfully.
[2025-04-18 04:08:31,893][__main__][INFO] - Starting trainer run...
[2025-04-18 04:08:31,893][src.trainers.base_trainer][INFO] - Initializing WandB run... Project: irve-hydra-refactor
[2025-04-18 04:08:33,117][src.trainers.base_trainer][INFO] - WandB Run Name: young-gorge-1, ID: pwp1yyf4
[2025-04-18 04:08:33,117][src.trainers.sae_trainer][INFO] - Setting up SAE Trainer components...
[2025-04-18 04:08:33,118][__main__][ERROR] - An error occurred during trainer instantiation or execution:
Traceback (most recent call last):
  File "/home/ozanbayiz/irve/src/main.py", line 44, in main
    trainer.run()
    ~~~~~~~~~~~^^
  File "/home/ozanbayiz/irve/src/trainers/sae_trainer.py", line 145, in run
    self._setup()       # Setup components
    ~~~~~~~~~~~^^
  File "/home/ozanbayiz/irve/src/trainers/sae_trainer.py", line 37, in _setup
    raise NotImplementedError(f"Data setup for {self.cfg.data._target_} not implemented in SAETrainer")
NotImplementedError: Data setup for src.datasets.datasets.FairFaceDataset not implemented in SAETrainer
[2025-04-18 04:08:33,119][__main__][ERROR] - Finishing WandB run with failure status due to error.
